### PR TITLE
Correctly skip PDO tests if required pdo_sqlite extension not enabled.

### PR DESCRIPTION
**Pull request by mprelude**
Previously, if `pdo_sqlite` was disabled, tests failed.

Changed behavior to skip these tests if required extension is not available:

```
PHPUnit 5.1.4 by Sebastian Bergmann and contributors.

..............SSSSSSSSS..SS...................................... 65 / 81 ( 80%)
................                                                  81 / 81 (100%)

Time: 623 ms, Memory: 10.00Mb

OK, but incomplete, skipped, or risky tests!
Tests: 81, Assertions: 132, Skipped: 11.
```

Tests still run correctly when `pdo_sqlite` is enabled:

```
PHPUnit 5.1.4 by Sebastian Bergmann and contributors.

................................................................. 65 / 81 ( 80%)
................                                                  81 / 81 (100%)

Time: 628 ms, Memory: 10.00Mb

OK (81 tests, 147 assertions)
```

### CODE REVIEW

File: tests/AuthFactoryTest.php [hunk 1]
```diff
@@ -29,6 +29,10 @@ public function testNewAuthWithSessionAndSegment()
 
     public function testNewPdoAdapter_passwordVerifier()
     {
+        if (false === extension_loaded('pdo_sqlite')) {
```
Comment (harikt) [inline tests/AuthFactoryTest.php]: How about we do a refactor to a trait ? 

$this->isExtensionLoaded('extension') ? and will it mark the test as skipped for it ?

In that case we could have saved some code.

Thoughts ?

Comment (mprelude) [inline tests/AuthFactoryTest.php]: I like that idea, will send a new commit at lunchtime or after work.

Thanks for the fast reply!

**Comment by mprelude** [comment]
@harikt, have refactored to a trait.

Travis wants to check 5.3 compatibility, but obviously traits weren't added until 5.4.

Since this functionality breaks 5.3 compatibility advertised in the readme, shall I reverse the commit?

**Comment by harikt** [comment]
As 5.3 has reached EOL http://php.net/supported-versions.php and this is only for the tests I believe that is ok for now.

I also think it may be good if we can change the `composer.json` for php version and travis.yml for the same.

Ping @pmjones if you have any suggestions / thoughts on this.

**Comment by harikt** [comment]
ping @brandonsavage for I believe you are also using the component.

**Comment by mprelude** [comment]
Updated composer.json.

**Comment by pmjones** [comment]
Changing the requirement to 5.4 is a BC break. :-/

**Comment by pmjones** [comment]
As such, that would mean bumping the major version number, which is fine if that's what needs to happen.

**Comment by mprelude** [comment]
@pmjones If I revert the last 2 commits it works on 5.3.

I'm not sure that the benefit of saving 6 LOC is worth breaking BC for.

**Comment by harikt** [comment]
@pmjones I think as 5.3 is already EOL and adding a constraint in composer.json to 5.4 will not let them install the newer version. So that may not break anything in their system and all can install upto the 5.3 releases. This may be minor release those.

If you still think that doesn't makes sense we may go back removing the last 2 commits.

Thank you

**Comment by harikt** [comment]
or may be as you said we could push this to 5.5 and v3 release also :)

**Comment by pmjones** [comment]
(/me grimaces)

I think the Right Thing To Do is keep the 5.3 requirement here, and start a 3.x branch for 5.5 or 5.6.

**Comment by pmjones** [comment]
However, @brandonsavage should weigh in as well.

**Comment by brandonsavage** [comment]
I agree that changing the min version is a major BC break which would require us to bump the version. That's okay, but I think we should do it as part and parcel of actually USING those features.

Also, I'd generally want to bump to a supported version of PHP, which would be 5.5.

**Comment by harikt** [comment]
Publishing a major version is not bad. We have chrome and firefox in the list ;) .

I am happy to push to 5.5 and 3.x .

**Comment by harikt** [comment]
3.x branch is being pushed. https://github.com/auraphp/Aura.Auth/tree/3.x .

Hopefully we can merge this PR to there.

**Comment by harikt** [comment]
I have fixed this via the cli to 3.x branch. If all is well, we can target to 3.x then .

https://github.com/auraphp/Aura.Auth/commits/3.x

**Comment by harikt** [comment]
Hi all,

I have cherry-picked the commit of @mprelude in https://github.com/auraphp/Aura.Auth/pull/82 .

Sorry for the delay and thank you for the PR.

Didn't noticed this long waiting PR.

**Pull request closed without merging**